sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -226,14 +226,19 @@ case class GenerateExec(
       "0"
     }
     val numOutput = metricTerm(ctx, "numOutputRows")
+    val requiredInput = {
+      input.zipWithIndex.flatMap { case (ev, i) =>
+        if ((parent.inputSet).contains(child.output(i))) Seq(ev) else Seq()
+      }
+    } ++ position ++ values
```

Comment:
more style nit: how about reformatting it ilke this?
```
    val requiredOutputVars = input.zip(child.output).filter(ev => parent.inputSet.contains(ev._2))
      .map(_._1) ++ position ++ values
```

Reply:
Done with slight modification.. Please check